### PR TITLE
Custom memory layout for STM32-L4R5ZI

### PR DESCRIPTION
**Pull request by jschneider-bensch**
We found out that, by default, the stack on L4R5ZI is linked into an insufficiently sized region of RAM. This leads to memory errors when the stack grows beyond that region. This PR makes us use a custom layout, which relocates the stack to a larger memory region.

The details of the memory layout for the L4R5ZI can be found in the [reference manual](https://www.st.com/resource/en/reference_manual/rm0432-stm32l4-series-advanced-armbased-32bit-mcus-stmicroelectronics.pdf) on page 94.

### CODE REVIEW

File: libcrux-nucleo-l4r5zi/memory.x [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+{
+    FLASH : ORIGIN = 0x08000000, LENGTH = 2048K /* BANK_1 */
+    RAM   : ORIGIN = 0x20000000, LENGTH =  640K /* SRAM */
+}
+
+
+/* This is where the call stack will be allocated. */
+/* The stack is of the full descending type. */
+/* You may want to use this variable to locate the call stack and static
+   variables in different memory regions. Below is shown the default value */
```
Comment (keks) [inline libcrux-nucleo-l4r5zi/memory.x:11]: Can you link the document that contains the notes here? https://md.cryspen.com/68n4HBCpR1qRe17_CRxiaQ?edit

**Review by keks — CHANGES_REQUESTED** [review]
Let's link the reason for having the custom linker script. Otherwise this looks good!

File: libcrux-nucleo-l4r5zi/memory.x (hunk 1 above)
Comment (jschneider-bensch) [reply libcrux-nucleo-l4r5zi/memory.x]: Good point! I've added a link there.

**Review by keks — APPROVED** [review]
Looks good!

**Pull request merged**